pkg/loki/modules.go

```diff
@@ -495,6 +499,60 @@ func (t *Loki) initCompactor() (services.Service, error) {
 	return t.compactor, nil
 }
 
+func (t *Loki) initDeleteRequestsStore() (serv services.Service, err error) {
+	if !t.cfg.PurgerConfig.Enable {
+		// until we need to explicitly enable delete series support we need to do create TombstonesLoader without DeleteStore which acts as noop
+		t.TombstonesLoader = purger.NewTombstonesLoader(nil, nil)
+
+		return
+	}
+
+	if loki_storage.UsingBoltdbShipper(t.cfg.SchemaConfig.SchemaConfig.Configs) {
+		return nil, errors.New("deletion feature is not supported with boltdb-shipper")
+	}
+
+	var indexClient chunk.IndexClient
+	reg := prometheus.WrapRegistererWith(
+		prometheus.Labels{"component": DeleteRequestsStore}, prometheus.DefaultRegisterer)
+	indexClient, err = storage.NewIndexClient(t.cfg.StorageConfig.DeleteStoreConfig.Store, t.cfg.StorageConfig.Config, t.cfg.SchemaConfig.SchemaConfig, reg)
+	if err != nil {
+		return
+	}
+
+	t.DeletesStore, err = purger.NewDeleteStore(t.cfg.StorageConfig.DeleteStoreConfig, indexClient)
+	if err != nil {
+		return
+	}
+
+	t.TombstonesLoader = purger.NewTombstonesLoader(t.DeletesStore, prometheus.DefaultRegisterer)
+
+	return
+}
+
+func (t *Loki) initPurger() (services.Service, error) {
+	if !t.cfg.PurgerConfig.Enable {
+		return nil, nil
+	}
+
+	storageClient, err := storage.NewObjectClient(t.cfg.PurgerConfig.ObjectStoreType, t.cfg.StorageConfig.Config)
+	if err != nil {
+		return nil, err
+	}
+
+	t.Purger, err = purger.NewPurger(t.cfg.PurgerConfig, t.DeletesStore, t.store, storageClient, prometheus.DefaultRegisterer)
+	if err != nil {
+		return nil, err
+	}
+
+	deleteRequestHandler := loki_purger.NewDeleteRequestHandler(t.DeletesStore, t.cfg.PurgerConfig.DeleteRequestCancelPeriod, prometheus.DefaultRegisterer)
+
+	t.server.HTTP.Path("/loki/api/admin/delete").Methods("PUT", "POST").Handler(t.httpAuthMiddleware.Wrap(http.HandlerFunc(deleteRequestHandler.AddDeleteRequestHandler)))
```

Comment:
I'm not sure about those path, I would have expected a DELETE method :) @slim-bean wdyt is that how cortex does it ?

Reply:
We should talk about this, it's very easy to change anyway, but it's better now than later.

Reply:
This is how Cortex does which is inspired from Prometheus. We can add support for DELETE method as well.

Reply:
Can you link me to Prometheus and Cortex docs please ?

Reply:
Prometheus: https://prometheus.io/docs/prometheus/latest/querying/api/#delete-series

Cortex: https://github.com/cortexproject/cortex/blob/6db67a4efbbf62b1133fa037a95382a21f752bbf/docs/guides/deleting-series.md

Reply:
Thanks I want to have a sense of the api request/responses.